### PR TITLE
SONARJAVA-2154 remove guava

### PR DESCRIPTION
**Pull request by benzonico**

### CODE REVIEW

File: java-frontend/src/main/java/org/sonar/java/ast/parser/ListTreeImpl.java [hunk 1]
```diff
@@ -71,44 +68,14 @@ public Kind kind() {
...
-    @Override
-    protected E computeNext() {
-      while (!iterables.isEmpty()) {
-        Iterator<? extends E> topIter = iterables.poll();
-        if (topIter.hasNext()) {
-          E result = topIter.next();
-          iterables.offer(topIter);
-          return result;
-        }
+    List<Tree> result = new ArrayList<>();
```
Comment (m-g-sonar) [inline java-frontend/src/main/java/org/sonar/java/ast/parser/ListTreeImpl.java]: :+1: x :100:

File: java-frontend/src/main/java/org/sonar/java/ast/parser/ArgumentListTreeImpl.java [hunk 2]
```diff
@@ -80,9 +79,16 @@ public SyntaxToken closeParenToken() {
...
   public Iterable<Tree> children() {
-    return Iterables.concat(
-      openParenToken != null ? Collections.singletonList(openParenToken) : Collections.<Tree>emptyList(),
-      super.children(),
-      closeParenToken != null ? Collections.singletonList(closeParenToken) : Collections.<Tree>emptyList());
+    List<Tree> result = new ArrayList<>();
+    if(openParenToken != null) {
+      result.add(openParenToken);
+    }
+    for (Tree tree : super.children()) {
```
Comment (m-g-sonar) [inline java-frontend/src/main/java/org/sonar/java/ast/parser/ArgumentListTreeImpl.java]: `super.children().forEach(res::add);` ?

File: java-frontend/src/main/java/org/sonar/java/model/DefaultJavaFileScannerContext.java [hunk 3]
```diff
@@ -145,25 +138,11 @@ public void reportIssue(JavaCheck javaCheck, Tree startTree, Tree endTree, Strin
 
   @Override
   public List<String> getFileLines() {
-    if(sonarComponents == null) {
```
Comment (m-g-sonar) [inline java-frontend/src/main/java/org/sonar/java/model/DefaultJavaFileScannerContext.java]: sonarComponents is still marked as being `@Nullable` in the constructor of the `DefaultJavaFileScannerContext`. Are you sure you can remove the null check?

Comment (benzonico) [inline java-frontend/src/main/java/org/sonar/java/model/DefaultJavaFileScannerContext.java]: look at file on master...

Comment (m-g-sonar) [inline java-frontend/src/main/java/org/sonar/java/model/DefaultJavaFileScannerContext.java]: Ok so maybe we should remove the `@Nullable` annotation.

File: java-checks/src/main/java/org/sonar/java/checks/DeprecatedHashAlgorithmCheck.java [hunk 4]
```diff
@@ -44,31 +44,33 @@
...
-    .put("getSha1Digest", SHA1)
-    .put("md5", MD5)
-    .put("md5Hex", MD5)
-    .put("sha1", SHA1)
-    .put("sha1Hex", SHA1)
-    .put("sha", SHA1)
-    .put("shaHex", SHA1)
-    .build();
+  private static final Map<String, String> ALGORITHM_BY_METHOD_NAME = new HashMap<>();
+    static {
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/DeprecatedHashAlgorithmCheck.java]: MAJOR Make this line start at column 3. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)

File: java-checks/src/main/java/org/sonar/java/checks/DeprecatedHashAlgorithmCheck.java [hunk 5]
```diff
@@ -44,31 +44,33 @@
...
+      ALGORITHM_BY_METHOD_NAME.put("sha", SHA1);
+      ALGORITHM_BY_METHOD_NAME.put("shaHex", SHA1);
+    }
 
   @Override
   protected List<MethodMatcher> getMethodInvocationMatchers() {
-    Builder<MethodMatcher> builder = ImmutableList.<MethodMatcher>builder()
-      .add(MethodMatcher.create()
+    List<MethodMatcher> builder = new ArrayList<>();
+      builder.add(MethodMatcher.create()
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/DeprecatedHashAlgorithmCheck.java]: MAJOR Make this line start at column 5. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)

File: java-checks/src/main/java/org/sonar/java/checks/DeprecatedHashAlgorithmCheck.java [hunk 6]
```diff
@@ -44,31 +44,33 @@
...
-    .put("md5Hex", MD5)
-    .put("sha1", SHA1)
-    .put("sha1Hex", SHA1)
-    .put("sha", SHA1)
-    .put("shaHex", SHA1)
-    .build();
+  private static final Map<String, String> ALGORITHM_BY_METHOD_NAME = new HashMap<>();
+    static {
+
+      ALGORITHM_BY_METHOD_NAME.put("getMd5Digest", MD5);
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/DeprecatedHashAlgorithmCheck.java]: MAJOR Make this line start at column 5. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 3 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Comment by m-g-sonar** [comment]
Closing the PR for the time being, as we can not remove our dependencies on guava before reworking some of our other dependencies relying on it.

**Pull request closed without merging**